src/SocialiteWasCalled.php

```diff
@@ -4,16 +4,30 @@
 class SocialiteWasCalled
 {
     /**
-     * @param string $providerName 'meetup'
-     * @param string $providerClass 'Your\Name\Space\ClassName'
+     * @param string $providerName   'meetup'
+     * @param string $providerClass  'Your\Name\Space\ClassNameProvider'
+     * @param string $providerServer 'Your\Name\Space\ClassNameServer'
      */
-    public function extendSocialite($providerName, $providerClass)
+    public function extendSocialite($providerName, $providerClass, $providerServer = null)
     {
-        $socialite = \App::make('Laravel\Socialite\Contracts\Factory');
+        $socialite = app()->make('Laravel\Socialite\Contracts\Factory');
+
         $socialite->extend(
             $providerName,
-            function ($app) use ($socialite, $providerName, $providerClass) {
-                $config = $app['config']['services.' . $providerName];
+            function ($app) use ($socialite, $providerName, $providerClass, $providerServer) {
+                $config = $app['config']['services.'.$providerName];
+
+                if (!empty($providerServer)) {
+                    return new $providerClass(
+                        $app['request'],
+                        new $providerServer([
```

Comment:
I don't think that we should define the actual makeup of this array here.  We could add a public helper method to this class but should allow the individual provider to provider to supply its own since it could be different.